File: frontend/lib/services/email_service.dart
import 'package:mailer/mailer.dart';
import 'package:mailer/smtp_server.dart';
import 'package:food_bridge/config/api_config.dart';

class EmailService {
  static final smtpServer = SmtpServer(
    ApiConfig.smtpHost,
    port: ApiConfig.smtpPort,
    username: ApiConfig.adminEmail,
    // Note: Password should be stored securely, preferably in environment variables
    password: const String.fromEnvironment('SMTP_PASSWORD'),
    ssl: false,
    allowInsecure: true,
  );

  static Future<void> sendVerificationEmail(String email, String name) async {
    final message = Message()
      ..from = Address(ApiConfig.adminEmail, 'Food Bridge Admin')
      ..recipients.add(email)
      ..subject = 'Your Charity Account has been Verified'
      ..html = '''
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #2196F3;">Congratulations!</h1>
          <p>Dear $name,</p>
          <p>Your charity account on Food Bridge has been verified. You can now start accepting donations and helping those in need.</p>
          <p>Here's what you can do now:</p>
          <ul>
            <li>Browse available donations in your area</li>
            <li>Accept donations that match your needs</li>
            <li>Coordinate pickup with donors</li>
            <li>Track your donation history</li>
          </ul>
          <p>If you have any questions or need assistance, please don't hesitate to contact our support team.</p>
          <p>Best regards,<br>Food Bridge Team</p>
          <div style="margin-top: 20px; padding-top: 20px; border-top: 1px solid #eee; font-size: 12px; color: #666;">
            <p>This is an automated message. Please do not reply to this email.</p>
          </div>
        </div>
      ''';

    try {
      await send(message, smtpServer);
    } catch (e) {
      throw Exception('Failed to send verification email: $e');
    }
  }

  static Future<void> sendDonationNotification({
    required String recipientEmail,
    required String recipientName,
    required String donationTitle,
    required String donorName,
    required String status,
  }) async {
    final message = Message()
      ..from = Address(ApiConfig.adminEmail, 'Food Bridge Admin')
      ..recipients.add(recipientEmail)
      ..subject = 'Donation $status: $donationTitle'
      ..html = '''
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #2196F3;">Donation Update</h1>
          <p>Dear $recipientName,</p>
          <p>The donation "$donationTitle" by $donorName has been $status.</p>
          <p>Please log in to your Food Bridge account to view more details and take necessary actions.</p>
          <p>Best regards,<br>Food Bridge Team</p>
          <div style="margin-top: 20px; padding-top: 20px; border-top: 1px solid #eee; font-size: 12px; color: #666;">
            <p>This is an automated message. Please do not reply to this email.</p>
          </div>
        </div>
      ''';

    try {
      await send(message, smtpServer);
    } catch (e) {
      throw Exception('Failed to send donation notification: $e');
    }
  }
} 

File: backend/services/emailService.js
const nodemailer = require('nodemailer');

class EmailService {
  constructor() {
    this.transporter = nodemailer.createTransport({
      host: process.env.SMTP_HOST,
      port: process.env.SMTP_PORT,
      secure: false,
      auth: {
        user: process.env.SMTP_USER,
        pass: process.env.SMTP_PASS
      }
    });
  }

  async sendVerificationEmail(email, name, verificationLink) {
    const mailOptions = {
      from: process.env.SMTP_USER,
      to: email,
      subject: 'Verify Your Food Bridge Charity Account',
      html: `
        <h1>Welcome to Food Bridge!</h1>
        <p>Dear ${name},</p>
        <p>Thank you for registering as a charity organization with Food Bridge. Your account is currently under review.</p>
        <p>We will notify you once your account has been verified by our admin team.</p>
        <p>Best regards,<br>The Food Bridge Team</p>
      `
    };

    return this.transporter.sendMail(mailOptions);
  }

  async sendDonationAcceptedEmail(donorEmail, donorName, charityName, donationTitle) {
    const mailOptions = {
      from: process.env.SMTP_USER,
      to: donorEmail,
      subject: 'Your Donation Has Been Accepted!',
      html: `
        <h1>Donation Accepted</h1>
        <p>Dear ${donorName},</p>
        <p>Your donation "${donationTitle}" has been accepted by ${charityName}.</p>
        <p>They will contact you shortly to arrange the pickup.</p>
        <p>Thank you for making a difference!</p>
        <p>Best regards,<br>The Food Bridge Team</p>
      `
    };

    return this.transporter.sendMail(mailOptions);
  }

  async sendCharityVerificationEmail(charityEmail, charityName) {
    const mailOptions = {
      from: process.env.SMTP_USER,
      to: charityEmail,
      subject: 'Your Food Bridge Account Has Been Verified!',
      html: `
        <h1>Account Verified</h1>
        <p>Dear ${charityName},</p>
        <p>Congratulations! Your Food Bridge charity account has been verified by our admin team.</p>
        <p>You can now start accepting donations through our platform.</p>
        <p>Thank you for joining our mission to reduce food waste and help those in need.</p>
        <p>Best regards,<br>The Food Bridge Team</p>
      `
    };

    return this.transporter.sendMail(mailOptions);
  }
}

module.exports = new EmailService(); 